PWGCF/Tasks/correlations.cxx

```diff
@@ -780,21 +806,29 @@ struct CorrelationTask {
     BinningTypeMCDerived configurableBinning{{axisVertex, axisMultiplicity}, true}; // true is for 'ignore overflows' (true by default). Underflows and overflows will have bin -1.
     auto tuple = std::make_tuple(mcParticles);
     SameKindPair<soa::Filtered<aod::CFMcCollisions>, soa::Filtered<aod::CFMcParticles>, BinningTypeMCDerived> pairs{configurableBinning, cfgNoMixedEvents, -1, mcCollisions, tuple, &cache}; // -1 is the number of the bin to skip
-
+    auto multiplicity = -1;
     for (auto it = pairs.begin(); it != pairs.end(); it++) {
       auto& [collision1, tracks1, collision2, tracks2] = *it;
       float eventWeight = 1.0f / it.currentWindowNeighbours();
 
+      if (cfgCentBinsForMC > 0) {
+        for (auto& collision : collisions) {
```

Comment:
This loop is wrong. You have to loop over the `groupedCollisions` from line 834. Otherwise you always loop over all collisions in the data frame.

When changing this also consider the comment above for the case when `size() == 0`, then we should skip STEP 0 (only when cfgCentBinsForMC > 0 of course)

Reply:
Ok. I tried to implement your suggestions. Please have a look at lines` 771` and `834`.